cub/test/insert_nested_NVTX_range_guard.h

```diff
@@ -29,7 +29,12 @@ struct NestedNVTXRangeGuard
   }
 };
 
-#  define CUB_DETAIL_BEFORE_NVTX_RANGE_SCOPE(name)                              \
-    ::cuda::std::optional<::NestedNVTXRangeGuard> __cub_nvtx3_reentrency_guard; \
-    NV_IF_TARGET(NV_IS_HOST, __cub_nvtx3_reentrency_guard.emplace(name););
+// TODO(giannis): Thrust algorithms lead to NVTX nesting, we still want to avoid nested ranges
+// on the CUB Device level side. This guard makes sure that when a newly added CUB primitive
+// uses another CUB primitive it calls it from the dispatch layer. We can disable the guard
+// just for thrust conditionally in the future, since thrust nesting is more often and very
+// frequent on the high API level.

```

Comment:
This should be tracked in an issue if it isn't already.

Reply:
indeed. will try to patch it after this one merges.

Reply:
https://github.com/NVIDIA/cccl/pull/4583